test/DeltaTriplesTest.cpp

```diff
@@ -0,0 +1,327 @@
+          vocab, localVocab,
+          {"<A> <B> <C>", "<B> <C> <D>", "<A> <low> <a>", "<B> <D> <C>"})));
+  EXPECT_THAT(
+      mapKeys(deltaTriples.triplesDeleted_),
+      testing::UnorderedElementsAreArray(makeIdTriples(
+          vocab, localVocab,
+          {"<A> <B> <D>", "<A> <B> <F>", "<A> <next> <B>", "<B> <next> <C>"})));
+
+  // Deleting unsorted triples.
+  deltaTriples.deleteTriples(
+      cancellationHandle,
+      makeIdTriples(vocab, localVocab, {"<C> <prev> <B>", "<B> <prev> <A>"}));
+  ASSERT_EQ(deltaTriples.numInserted(), 4);
+  ASSERT_EQ(deltaTriples.numDeleted(), 6);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(10));
+  EXPECT_THAT(
+      mapKeys(deltaTriples.triplesInserted_),
+      testing::UnorderedElementsAreArray(makeIdTriples(
+          vocab, localVocab,
+          {"<A> <B> <C>", "<B> <C> <D>", "<A> <low> <a>", "<B> <D> <C>"})));
+  EXPECT_THAT(mapKeys(deltaTriples.triplesDeleted_),
+              testing::UnorderedElementsAreArray(makeIdTriples(
+                  vocab, localVocab,
+                  {"<A> <B> <D>", "<A> <B> <F>", "<A> <next> <B>",
+                   "<B> <next> <C>", "<C> <prev> <B>", "<B> <prev> <A>"})));
+
+  // Deleting previously deleted triples.
+  deltaTriples.deleteTriples(
+      cancellationHandle, makeIdTriples(vocab, localVocab, {"<A> <next> <B>"}));
+  ASSERT_EQ(deltaTriples.numInserted(), 4);
+  ASSERT_EQ(deltaTriples.numDeleted(), 6);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(10));
+  EXPECT_THAT(
+      mapKeys(deltaTriples.triplesInserted_),
+      testing::UnorderedElementsAreArray(makeIdTriples(
+          vocab, localVocab,
+          {"<A> <B> <C>", "<B> <C> <D>", "<A> <low> <a>", "<B> <D> <C>"})));
+  EXPECT_THAT(mapKeys(deltaTriples.triplesDeleted_),
+              testing::UnorderedElementsAreArray(makeIdTriples(
+                  vocab, localVocab,
+                  {"<A> <B> <D>", "<A> <B> <F>", "<A> <next> <B>",
+                   "<B> <next> <C>", "<C> <prev> <B>", "<B> <prev> <A>"})));
+
+  // Inserting previously deleted triple.
+  deltaTriples.insertTriples(cancellationHandle,
+                             makeIdTriples(vocab, localVocab, {"<A> <B> <F>"}));
+  ASSERT_EQ(deltaTriples.numInserted(), 5);
+  ASSERT_EQ(deltaTriples.numDeleted(), 5);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(10));
+  EXPECT_THAT(mapKeys(deltaTriples.triplesInserted_),
+              testing::UnorderedElementsAreArray(
+                  makeIdTriples(vocab, localVocab,
+                                {"<A> <B> <C>", "<A> <B> <F>", "<B> <C> <D>",
+                                 "<A> <low> <a>", "<B> <D> <C>"})));
+  EXPECT_THAT(mapKeys(deltaTriples.triplesDeleted_),
+              testing::UnorderedElementsAreArray(makeIdTriples(
+                  vocab, localVocab,
+                  {"<A> <B> <D>", "<A> <next> <B>", "<B> <next> <C>",
+                   "<C> <prev> <B>", "<B> <prev> <A>"})));
+}
```

Comment:
Once you have useful matchers for this testing, the tested cases become visible.